lib/logstash/outputs/elasticsearch/elasticsearch_configs.rb

```diff
@@ -1,15 +1,23 @@
-require 'forwardable' # Needed for logstash core SafeURI. We need to patch this in core: https://github.com/elastic/logstash/pull/5978
-
 module LogStash; module Outputs; class ElasticSearch
-  module CommonConfigs
+  module ElasticsearchConfigs
```

Comment:
If all of these are only ever used in `LogStash::Outputs::ElasticSearch`, is there value in having them extracted to `LogStash::Outputs::ElasticSearch::ElasticsearchConfigs`? Should we be doubling-down on removing them from directly being on the output plugin's class, or should the ones that only apply to the output be defined on the output?

Reply:
Good point. So my thinking was that we already had that `CommonConfigs` module which defined *some* options while others were defined in the plugin class and I did not really see any clear indications on the options separation choices, there is certainly some history here. My goad is to extract what will be shared with the data streams output which will depend on this output. I decided to simply put all options which were specific to this output into the ElasticsearchConfigs module and all other which I think will be shared with data streams into the SharedConfigs module. 

I am +1 on having these options back into the plugin class. There is no particular need to have these into a module.